src/madness/chem/potentialmanager.cc

```diff
@@ -135,8 +136,41 @@ double WignerSeitzPotentialFunctor::operator()(const coord_3d &r) const {
       }
     }
   }
-
   return result;
 }
 
+SAPFunctor::SAPFunctor(
+    const Atom &atom,
+    const BoundaryConditions<3> &bc,
+    const Tensor<double> &cell,
+    int special_level)
+    : atom(atom), bc_(bc), cell(cell), special_level_(special_level),
+      special_points_({atom.get_coords()}) {
+}
+
+double SAPFunctor::operator()(const coord_3d &x) const {
+  const double zero_cutoff = 1e-10;
```

Comment:
I am not sure why you need this

Reply:
I could well be not understanding some numerical analysis here, but if `r = 0`, l. 168 may try to compute `0 / 0`, no?

Reply:
Yes, for sure; yet, I thought that the multiresolution scheme does not put any points at the nucleus. Still, I think `1e-10` is a hardcoded cutoff that is somewhat meaningless; it could also be `DBL_EPSILON` or a smaller number since its reciprocal is still a normal number.

Reply:
@susilehtola when projecting to MW basis this functor will be called for points whose coordinates are controlled by the Gauss-Legendre quadrature so unlikely that you will hit the nucleus exactly, but must consider this as a possibility

Reply:
Yes but like you point out below, MADNESS doesn't use the Coulomb potential but the smoothed potential that doesn't have this issue, so the cutoff can be removed altogether...